new-log-viewer/src/utils/config.ts

```diff
@@ -0,0 +1,128 @@
+import {PAGE_SIZE} from "../contexts/StateContextProvider";
+import {
+    CONFIG_NAME,
+    ConfigMap,
+    ConfigUpdate,
+    LOCAL_STORAGE_KEY,
+} from "../typings/config";
+
+
+const DECODER_OPTIONS_DEFAULT = {
+    formatString: "%d{yyyy-MM-dd HH:mm:ss.SSS} [%process.thread.name] %log.level" + " %message%n",
+    logLevelKey: "log.level",
+    timestampKey: "@timestamp",
+};
+const THEMES = ["light",
+    "dark"];
+const THEME_DEFAULT = "light";
+
+/**
+ * Retrieves the configuration value from local storage based on the provided configuration name.
+ *
+ * @template T - The type of the configuration name.
+ * @param code The configuration name to retrieve the value for.
+ * @return - The configuration value or null if not found.
+ */
+const getConfig = <T extends CONFIG_NAME>(code: T): ConfigMap[T] | null => {
```

Comment:
I feel it might reduce the cognitive load if we can write
```
const getConfig = <T extends CONFIG_NAME>(name: T): ConfigMap[T] | null => {
```

or even better - we can rename `CONFIG_NAME` -> `CONFIG_CODE`.